test/NameBinding/import-cross-import-loading.swift

```diff
@@ -0,0 +1,128 @@
+// RUN: not %target-typecheck-verify-swift -enable-cross-import-overlays -I %t/include -I %t/lib/swift -F %t/Frameworks -DDONT_ALWAYS_IMPORT -DFAT_LIBRARY 2>/dev/null
+// RUN: not %target-typecheck-verify-swift -enable-cross-import-overlays -I %t/include -I %t/lib/swift -F %t/Frameworks -DDONT_ALWAYS_IMPORT -DCLANG_LIBRARY 2>/dev/null
+// RUN: not %target-typecheck-verify-swift -enable-cross-import-overlays -I %t/include -I %t/lib/swift -F %t/Frameworks -DDONT_ALWAYS_IMPORT -DSWIFT_FRAMEWORK 2>/dev/null
+// RUN: not %target-typecheck-verify-swift -enable-cross-import-overlays -I %t/include -I %t/lib/swift -F %t/Frameworks -DDONT_ALWAYS_IMPORT -DCLANG_FRAMEWORK 2>/dev/null
+
+//
+// Actual test code:
+//
+
+// We should have loaded _AlwaysImportedOverlay.
+fromAlwaysImportedOverlay() // no-error
+fromAlwaysImportedPlatformOverlay() // no-error
+
+// We should *not* have loaded _NeverImportedOverlay
+fromNeverImportedOverlay() // expected-error {{use of unresolved identifier 'fromNeverImportedOverlay'}}
+
+// Unlike real cross-import overlays, our test overlay doesn't @_exported import
+// the underlying module, so you can't call underlying declarations at all in
+// this test.
+//
+// The fact that we cannot see these declarations in this test indicates that,
+// if we were using a well-formed overlay, the overlay could shadow them if it
+// chose.
+
+fromThinLibrary() // expected-error {{use of unresolved identifier 'fromThinLibrary'}}
+fromFatLibrary() // expected-error {{use of unresolved identifier 'fromFatLibrary'}}
+fromClangLibrary() // expected-error {{use of unresolved identifier 'fromClangLibrary'}}
+fromSwiftFramework() // expected-error {{use of unresolved identifier 'fromSwiftFramework'}}
+fromClangFramework() // expected-error {{use of unresolved identifier 'fromClangFramework'}}
+
+//
+// FileCheck output for -emit-dependencies:
+//
+
+// DEPS-LABEL: - :
+
+// DEPS-NOT: lib{{/|\\}}swift{{/|\\}}_NeverImportedOverlay.swiftinterface
+// DEPS-NOT: badarch-badvendor-bados
+
+// DEPS-DAG: lib{{/|\\}}swift{{/|\\}}_AlwaysImportedOverlay.swiftinterface
+// DEPS-DAG: lib{{/|\\}}swift{{/|\\}}_AlwaysImportedPlatformOverlay.swiftinterface
+// DEPS-DAG: [[MODULE]].swiftcrossimport{{/|\\}}AlwaysImported.swiftoverlay
+// DEPS-DAG: [[MODULE]].swiftcrossimport{{/|\\}}[[TARGET]]{{/|\\}}AlwaysImported.swiftoverlay
+
+// FIXME: Better if we didn't?
+// DEPS-DAG: [[MODULE]].swiftcrossimport{{/|\\}}NeverImported.swiftoverlay
+
+//
+// Imports used in certain test cases:
+//
+
+#if !ALWAYS_IMPORTED_LAST && !DONT_ALWAYS_IMPORT
+import AlwaysImported
+#endif
+
+fromAlwaysImported()
+
+#if THIN_LIBRARY
+import ThinLibrary
+#endif
+
+#if FAT_LIBRARY
+import FatLibrary
+#endif
+
+#if CLANG_LIBRARY
+import ClangLibrary
+#endif
+
+#if SWIFT_FRAMEWORK
+import SwiftFramework
+#endif
+
+#if CLANG_FRAMEWORK
+import ClangFramework
+#endif
```

Comment:
I should have test variants with a clang overlay—one where the clang module declares the cross-import overlays, another where the clang overlay does.

Reply:
Done. (Using one test where both the Swift overlay and clang module declare a cross-import overlay.)